src/convert.jl

```diff
@@ -248,15 +248,40 @@ Converts a hyperrectangular set to a zonotope.
 
 ### Input
 
-- `Zonotope`
-- `H` -- hyperrectangular set
+- `Zonotope` -- type, used for dispatch
+- `H`        -- hyperrectangular set
 
 ### Output
 
 A zonotope.
 """
-function convert(::Type{Zonotope}, H::AbstractHyperrectangle)
-    return Zonotope(center(H), Diagonal(radius_hyperrectangle(H)))
+function convert(::Type{Zonotope}, H::AbstractHyperrectangle{N}) where {N}
+    if isflat(H)
+        r = radius_hyperrectangle(H)
+        n = length(r)
+
+        nzgen = 0
+        Gnz = Vector{N}()
+        sizehint!(Gnz, n * n)
+        @inbounds for (i, ri) in enumerate(r)
+            if ri != zero(N)
```

Comment:
Do you want to do an approximate zero check here?

Reply:
maybe, i have t think about it. not in this PR.